frame/bridge/relayer-game/src/lib.rs

```diff
@@ -419,12 +419,24 @@ decl_module! {
 				<Headers<T, I>>::remove_prefix(game_id);
 				<Proposals<T, I>>::take(game_id);
 			};
-			let update_samples = |game_id, chain_len| {
-				<Samples<T, I>>::mutate(game_id, |samples|
-					T::RelayerGameAdjustor::update_samples(
-						T::RelayerGameAdjustor::round_from_chain_len(chain_len) + 1,
-						samples
-					)
+			let update_samples = |game_id| {
+				<Samples<T, I>>::mutate(game_id, |samples| {
+						T::RelayerGameAdjustor::update_samples(samples);
+
+						if samples.len() < 2 {
+							error!("Sample Points MISSING, \
+								Check Your Sample Strategy Implementation");
+
+							return;
+						}
+
+						Self::deposit_event(RawEvent::NewRound(
+							game_id,
+							samples.concat(),
+							samples[samples.len() - 1].clone(),
```

Comment:
@AurevoirXavier 
This works for ethereum relay samples, but may not apply to other relays.

The following does not necessarily always true:

```
samples[samples.len() - 1].iter().max <= samples[samples.len() - 2].iter().max()
```